agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -108,12 +86,11 @@ def modify_create_opts(opts)
     # @param [Hash] opts
     def modify_host_config(opts)
       host_config = opts['HostConfig'] || {}
-      host_config['VolumesFrom'] ||= []
-      host_config['VolumesFrom'] << "weavewait-#{WEAVE_VERSION}:ro"
       dns = interface_ip('docker0')
       if dns && host_config['NetworkMode'].to_s != 'host'
         host_config['Dns'] = [dns]
         host_config['DnsSearch'] = ['kontena.local']
+        host_config['DnsOptions'] = ['use-vc'] # tcp mode for dns lookups
```

Comment:
with Docker overlay network the DNS goes always through Docker DNS layer (`nameserver 127.0.0.11` always added to container resolv.conf). While testing this we noticed with @jakolehm that there were lots of timeouts and errors in udp mode. Changing to tcp mode works much better.